src/services/bunker.py

```diff
@@ -1,60 +1,26 @@
-from typing import Any, Mapping
-
-import math
 import logging
-from collections import defaultdict
-from dataclasses import dataclass
 from functools import lru_cache
 
-from web3.types import Wei, EventData
-
-from src.constants import (
-    PROPORTIONAL_SLASHING_MULTIPLIER_BELLATRIX,
-    EFFECTIVE_BALANCE_INCREMENT,
-    EPOCHS_PER_SLASHINGS_VECTOR,
-    MIN_VALIDATOR_WITHDRAWABILITY_DELAY,
-    MAX_EFFECTIVE_BALANCE,
-    GWEI_TO_WEI,
-    TOTAL_BASIS_POINTS,
-)
-from src.providers.keys.typings import LidoKey
-from src.utils.slot import get_first_non_missed_slot
+from src.constants import TOTAL_BASIS_POINTS, GWEI_TO_WEI
+from src.services.bunker_cases.abnormal_cl_rebase import AbnormalClRebase
+from src.services.bunker_cases.midterm_slashing_penalty import MidtermSlashingPenalty
 
-from src.modules.accounting.typings import Gwei, LidoReportRebase
+from src.modules.accounting.typings import LidoReportRebase
 from src.modules.submodules.consensus import FrameConfig, ChainConfig
 from src.providers.consensus.typings import Validator
-from src.typings import BlockStamp, SlotNumber, EpochNumber, BlockNumber, ReferenceBlockStamp
+from src.services.bunker_cases.typings import BunkerConfig
+from src.typings import BlockStamp, ReferenceBlockStamp, Gwei
 from src.web3py.extensions.lido_validators import LidoValidator
 from src.web3py.typings import Web3
 
 
 logger = logging.getLogger(__name__)
 
 
-@dataclass
-class BunkerConfig:
-    normalized_cl_reward_per_epoch: int
-    normalized_cl_reward_mistake_rate: float
-    rebase_check_nearest_epoch_distance: int
-    rebase_check_distant_epoch_distance: int
-
-
 class BunkerService:
     """
     https://research.lido.fi/t/withdrawals-for-lido-on-ethereum-bunker-mode-design-and-implementation/
     """
-    b_conf: BunkerConfig
-    c_conf: ChainConfig
-    f_conf: FrameConfig
-
-    simulated_cl_rebase: LidoReportRebase
-
-    last_report_ref_slot: SlotNumber = SlotNumber(0)
-
-    all_validators: dict[str, Validator] = {}
-    lido_keys: dict[str, LidoKey] = {}
-    lido_validators: dict[str, LidoValidator] = {}
-
     def __init__(self, w3: Web3):
         self.w3 = w3
 
```

Comment:
Let's move cache to method in accounting

Reply:
done